File: src/main/scala/qmf/poc/agent/catalog/models/ObjectDirectory.scala
package qmf.poc.agent.catalog.models

import spray.json.{JsNumber, JsObject, JsString, JsValue, JsonFormat}

import java.util.Date

class ObjectDirectory(
    val owner: String,
    val name: String,
    val `type`: String,
    val subType: String,
    val objectLevel: Int,
    val restricted: String,
    val model: String,
    val created: String,
    val modified: String,
    val lastUsed: String
)
object ObjectDirectory:
  given JsonFormat[ObjectDirectory] with
    def write(o: ObjectDirectory): JsObject =
      JsObject(
        "name" -> JsString(o.name),
        "owner" -> JsString(o.owner),
        "type" -> JsString(o.`type`),
        "objectLevel" -> JsNumber(o.objectLevel),
        "subType" -> JsString(o.subType),
        "model" -> JsString(o.model),
        "created" -> JsString(o.created),
        "lastUser" -> JsString(o.lastUsed),
        "modified" -> JsString(o.modified),
        "restricted" -> JsString(o.restricted)
      )

    def read(value: JsValue): ObjectDirectory = ???

File: src/main/java/qmf/poc/agent/catalog/models/ObjectDirectory.java
package qmf.poc.agent.catalog.models;

public class ObjectDirectory {
    public final String owner;
    public final String name;
    public final String type;
    public final String subType;
    public final int objectLevel;
    public final String restricted;
    public final String model;
    public final String created;
    public final String modified;
    public final String lastUsed;

    public ObjectDirectory(String owner, String name, String type, String subType, int objectLevel,
                           String restricted, String model, String created, String modified, String lastUsed) {
        this.owner = owner;
        this.name = name;
        this.type = type;
        this.subType = subType;
        this.objectLevel = objectLevel;
        this.restricted = restricted;
        this.model = model;
        this.created = created;
        this.modified = modified;
        this.lastUsed = lastUsed;
    }

    @Override
    public String toString() {
        String s = "'" + owner + "'|'" + name + "'|'" + type + "'|'" + subType + "'|'" + objectLevel +
                "'|'" + restricted + "'|'" + model + "'|'" + created + "'|'" + modified + "'|'" + lastUsed;// + "'";
        return s.length() <= 131 ? s + "'" : s.substring(0, 129) + "...'";
    }
}
